rosbag2_to_video/api.py

```diff
@@ -48,120 +62,256 @@ def add_arguments_to_parser(argparser):
         required=True,
         help='Output filename (currently only supports MP4)'
     )
-    argparser.add_argument('--fps', type=int, required=True, help='Output frames per second')
-    argparser.add_argument('--storage-id', type=str, default="sqlite3", help='Rosbag2 storage id')
-    argparser.add_argument('--codec', type=str, default="avc1", help='Video codec')
+    argparser.add_argument('--fps', type=float, required=True, help='Output frames per second')
+    argparser.add_argument(
+        '--storage-id',
+        type=str,
+        default='sqlite3',
+        help='Rosbag2 storage id. If a bag folder is provided, this is ignored.')
+    argparser.add_argument(
+        '--codec', type=cv2_video_writer_fourcc, default="avc1", help='Video codec')
 
 
-def convert_bag_to_video(args):
-    # Force the .mp4 extension on file output name
-    if args.output[-4:] != '.mp4':
-        args.output += '.mp4'
-    if os.path.isdir(args.bagfile):
+class CommandInputError(ValueError):
+    """Error raised when provided command line arguments are not valid."""
+
+    def __init__(self, msg: str):
+        super().__init__(msg)
+
+
+def get_stamp_from_image_msg(image_msg) -> float:
+    """Convert timestamp in msg from nanoseconds to seconds."""
+    stamp = rclpy.time.Time.from_msg(image_msg.header.stamp).nanoseconds
+    stamp = stamp / 1e9
+    return stamp
+
+
+def get_topic_type(topic_name: str, topics_and_types) -> str:
+    """Get the topic type from the topic name and the topic information in the bag."""
+    try:
+        topic_type = next(x for x in topics_and_types if x.name == topic_name).type
+    except StopIteration:
+        raise CommandInputError(
+            f'Topic {topic_name} was not recorded in the bagfile')
+        return None
```

Comment:
We're raising so the return isn't necessary

Reply:
Done in https://github.com/ivanpauno/rosbag2_to_video/pull/5